### ISSUE TITLE
Problems with absolute paths in plots under LaTeX (Windows)

### ISSUE THREAD

**Issue by NMarkgraf** [rstudio/rmarkdown#1289]
Hi Yihui,

well, me again, I got into trouble with the handling of the rmardown generated plots in LaTeX.

As you might now, we are using Rmarkdown for a little Project with many writes on different operating systems. So our scripts have to run under Windows and *nix systems. But there is a little problem with the handling of plots.

I've made a short example, with is include in this bug report, but I will explain it  first:

Our directory structure is like this:

```
+-content/
!   +-child.Rmd
+-images/
!  +-rmarkdowrn.png
+-build/
+-MainTestDocument.Rmd
+-Makefile.R
```

We use the Makefile.R to build MainTestDocument.Rmd as MTDocument.pdf (as a beamer presentation) in the build/ directory.

We got no problem with including images, like rmarkdown in the images/ directory, because they will include with a relative path. 

But all plots will have a absolute path, with causes Windows system to get in trouble because of the absolute path part to the main document directory. 

We got something like

```tex
\includegraphics{C:\path\to\main\document\dirctory/path/to/generated/images.png}
```

On *nix systems it'll be:
```tex
\includegraphics{/path/to/main/document/dirctory/path/to/generated/images.png}
```

which is also not nice, but will do it. But it would be much nice to have only relative paths or correct paths on windows systems.

Thank in advance,
Norman

BTW: If I use the kntir-button in RStudio it works fine.

[yihui-test.zip](https://github.com/rstudio/rmarkdown/files/1828445/yihui-test.zip)

**Comment by yihui** [comment]
I'm sure this issue has been reported at least once before, although I don't remember which issue it was. Usually I don't recommend using `output_dir` or `output_file` or `intermediate_dir`. They are known to be problematic, e.g. https://github.com/rstudio/rmarkdown/issues/587#issuecomment-168437646 #238 #348 #574 #1033 #861 etc. I believe the only reliable strategy is to render everything in the current directory, and move the output files later.

In short, yes, I admit this is a bug but I'm afraid it will take very long to fix.

**Comment by NMarkgraf** [comment]
Sadly, @yihui, this is not an option for us. There must be a place where the absolute path is added, and the only thing needed there is to replace all "\" by "/". - That will take to long? If you tell me where, I will try to fix it. ... :-(

**Comment by NMarkgraf** [comment]
Okay @yihui,

after looking at the code I will ask you to give the following a try:

in render.R starting at line 363 you have got the following:

```
    # use filename based figure and cache directories
    base_pandoc_to <- gsub('[-+].*', '', pandoc_to)
    if (base_pandoc_to == 'html4') base_pandoc_to <- 'html'
    figures_dir <- paste(files_dir, "/figure-", base_pandoc_to, "/", sep = "")
    knitr::opts_chunk$set(fig.path = figures_dir)
    cache_dir <- knitr_cache_dir(input, base_pandoc_to)
    knitr::opts_chunk$set(cache.path = cache_dir)
```

I guess replacing this by:

```
    # use filename based figure and cache directories
    base_pandoc_to <- gsub('[-+].*', '', pandoc_to)
    if (base_pandoc_to == 'html4') base_pandoc_to <- 'html'
    loc_file_dir <- file.path(output_dir, knitr_files_dir(basename(output_file)))
    figures_dir <- paste0(file.path(loc_files_dir, "figure-", base_pandoc_to), "/")
    # may be convert it with pandoc_path_args 
    #    figures_dir <- pandoc_path_args(paste0(file.path(loc_files_dir, "figure-", base_pandoc_to), "/"), FALSE)
    knitr::opts_chunk$set(fig.path = figures_dir)
    cache_dir <- knitr_cache_dir(input, base_pandoc_to)
    knitr::opts_chunk$set(cache.path = cache_dir)
```

will make it. I'm not sure if it should be converted by pandoc_path_args also, but I think the rest would be perfect. If I am right, this is the point, where the "c:\" is included, because files_dir is converted first to "c:\" and used mixed up with the "/" stuff. If you keep it in slash mode ("/") instead of mixing backslash an slash notation, I guess this might help a lot.

Yours
Norman

**Comment by NMarkgraf** [comment]
Okay, next step.

I created a fork, tried it on a *nix and a windows system and it worked!

Here is the pull request for this fix https://github.com/rstudio/rmarkdown/pull/1293 

Please take a minute and review it, and maybe include it. - Thanks, and keep on doing the great stuff.

**Comment by NMarkgraf** [comment]
Worked for me. - Hope your solution won't break any other thing ...

**Comment by NMarkgraf** [comment]
Here we are .. the problem is BACK AGAIIN … Why?
